src/RA_Defs.h

```diff
@@ -225,30 +255,382 @@ typedef DWORD			ARGB;
 #undef ASSERT
 #define ASSERT( x ) {}
 #endif
-	
+
 #ifndef UNUSED
 #define UNUSED( x ) ( x );
 #endif
 
-extern std::string Narrow(const wchar_t* wstr);
-extern std::string Narrow(const std::wstring& wstr);
-extern std::wstring Widen(const char* str);
-extern std::wstring Widen(const std::string& str);
 
-//	No-ops to help convert:
-//	No-ops to help convert:
-extern std::wstring Widen(const wchar_t* wstr);
-extern std::wstring Widen(const std::wstring& wstr);
-extern std::string Narrow(const char* str);
-extern std::string Narrow(const std::string& wstr);
 
-typedef std::basic_string<TCHAR> tstring;
 
 
+
+
+
+namespace ra {
+using cstring       = const char*;
+using cwstring      = const wchar_t*;
+using tstring       = std::basic_string<TCHAR>;
+
+template<typename CharT>
+using string_t = std::basic_string<CharT>;
+
+using DataStream     = std::basic_string<BYTE>;
+using ustring        = DataStream;
+using uofstream      = std::basic_ofstream<BYTE>;
+using uifstream      = std::basic_ifstream<BYTE>;
+using uostringstream = std::basic_ostringstream<BYTE>;
+using uistringstream = std::basic_istringstream<BYTE>;
+using uostream       = std::basic_ostream<BYTE>;
+using uistream       = std::basic_istream<BYTE>;
+
+using ByteAddress   = std::size_t;
+using DataPos       = std::size_t;
+using AchievementID = std::size_t;
+using LeaderboardID = std::size_t;
+using GameID        = std::size_t;
+
+// forgot to add this
+// if you are already in ra or type "using namespace ra" that's good enough
+inline namespace int_literals
+{
+
+// Don't feel like casting non-stop
```

Comment:
These feel dirty. More "magic" code, and most of them will never be used.

Reply:
Well there's a few literals that don't exist. size_t, inptr_t/ptrdiff_t, DWORD (it has a variable size like size_t, it gave me warnings before somewhere), short, unsigned short. It would be a lot simpler to just use a literal instead of a static cast when that can already be done. A lot of them will be used. Implicit conversions are probably the the second if not first worst problem in programming.

Reply:
>A lot of them will be used.

Can we not add them until they are?

Reply:
Sure